src/Symfony/Component/Webhook/Server/JsonBodyConfigurator.php

```diff
@@ -20,14 +20,22 @@
  */
 final class JsonBodyConfigurator implements RequestConfiguratorInterface
 {
-    public function __construct(
-        private readonly SerializerInterface $serializer,
-    ) {
+    private PayloadEncoderInterface $payloadEncoder;
+
+    public function __construct(SerializerInterface|PayloadEncoderInterface|null $payloadEncoder = null)
+    {
+        if (null === $payloadEncoder) {
+            $this->payloadEncoder = new NativeJsonPayloadEncoder();
+        } elseif ($payloadEncoder instanceof SerializerInterface) {
+            $this->payloadEncoder = new SerializerPayloadEncoder($payloadEncoder);
+        } else {
+            $this->payloadEncoder = $payloadEncoder;
+        }
```

Comment:
Can we compute all these conditionals externally and inject the appropriate service instead? This would also remove the null type in the constructor and the notion that this configurator requires an optional encoder

Reply:
yes indeed, good idea 👍